pkg/provisioner/ironic/ironic.go

```diff
@@ -420,65 +420,21 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged bool) (r
 			imageData = p.host.Spec.Image
 		}
 
-		checksum, checksumType, ok := imageData.GetChecksum()
-
+		_, _, ok := imageData.GetChecksum()
 		if ok {
-			p.log.Info("setting instance info",
-				"image_source", imageData.URL,
-				"image_os_hash_value", checksum,
-				"image_os_hash_algo", checksumType,
-			)
-
-			updates := nodes.UpdateOpts{
-				nodes.UpdateOperation{
-					Op:    nodes.AddOp,
-					Path:  "/instance_info/image_source",
-					Value: imageData.URL,
-				},
-				nodes.UpdateOperation{
-					Op:    nodes.AddOp,
-					Path:  "/instance_info/image_os_hash_value",
-					Value: checksum,
-				},
-				nodes.UpdateOperation{
-					Op:    nodes.AddOp,
-					Path:  "/instance_info/image_os_hash_algo",
-					Value: checksumType,
-				},
+			updates, err := p.getImageUpdateOptsForNode(ironicNode, imageData)
+			if err != nil {
+				return result, errors.Wrap(err, "Could not get Image options for node")
+			}
+			updates = append(
+				updates,
 				nodes.UpdateOperation{
 					Op:    nodes.ReplaceOp,
 					Path:  "/instance_uuid",
 					Value: string(p.host.ObjectMeta.UID),
 				},
```

Comment:
I was picturing that this would be set inside `getImageUpdateOptsForNode()`. It's a thing that always needs to be set at the same time as the `/instance_info` (i.e. before provisioning or adopting), so there's no reason not to keep all the related code together.
Now that I write that, it's suspicious that RootGB is the only `/instance_info` key that is not set in `getImageUpdateOptsForNode()` but left behind in `getUpdateOptsForNode()` (which possibly also suggests another name for this function). I suspect that the reason it hasn't broken adoption is that the [issue in Ironic](https://storyboard.openstack.org/#!/story/2005165) referenced in the comments has loooooong since been fixed, and we should just delete it altogether as thoroughly obsolete (pretty sure we already rely on other Ironic behaviour that is less than 2 years old).

Reply:
Yeah I was going to add this to `getImageUpdateOptsForNode()` but then noticed is that it's already set in `getImageUpdateOptsForNode()` and we only pass the `p.host.Spec.Image` into that function - if you prefer I can rework the parameters so we can set instance_uuid there, and add a commit to remove the `root_gb`

Reply:
Thanks, I moved `instance_info` to `getImageUpdateOptsForNode` - I'll propose a separate PR with the removal of `root_gb`